src/main/java/carpet/script/api/Scoreboards.java

```diff
@@ -133,8 +136,13 @@ public static void apply(Expression expression)
             }
 
             ScoreboardObjective objective = scoreboard.getObjective(objectiveName);
-            if (objective != null)
+            if (objective != null) {
+                ((Scoreboard_scarpetMixin)scoreboard).getObjectivesByCriterion().get(objective.getCriterion()).remove(objective);
```

Comment:
Yeah, I would check if the criterion is same and skip doing anything, returning `null` for a change.

Reply:
Yes that makes a lot of sense! I was also wondering, when creating an objective without specifying a criterion, it defaults to dummy. But should it also default to dummy when changing a criterion?
I can imagine the scenario of someone just running `scoreboard_add(objective)` with an already existing objective by accident, and not realizing that the criterion got reset to dummy. Maybe that should only change when a criterion is actually specified.

In that case, running the above could return the criterion as a string maybe?
It wouldnt fir the name of the function, as well as scoreboard_set if it got changed though